ToyBox/classes/MonkeyPatchin/BagOfPatches/Alignment.cs

```diff
@@ -58,12 +58,9 @@ private static class UnitAlignment_Shift_Patch {
                         if (direction == AlignmentShiftDirection.TrueNeutral && (Vector2.zero - __instance.m_Vector).magnitude < num) {
                             newAlignment = Vector2.zero;
                         }
-                        Traverse.Create(__instance).Property<Vector2>("Vector").Value = newAlignment;
-                        Traverse.Create(__instance).Method("UpdateValue").GetValue();
-                        //Traverse requires the parameter types to find interface parameters
-                        Traverse.Create(__instance).Method("OnChanged",
-                            new Type[] { typeof(AlignmentShiftDirection), typeof(Vector2), typeof(IAlignmentShiftProvider), typeof(bool) },
-                            new object[] { direction, vector, provider, true }).GetValue();
+                        __instance.SetVector(newAlignment);
```

Comment:
Note: this will respect alignment locks and correct for them, though I don't know if there are any situations where MC can get alignment locked outside of Toybox itself. If we want it to ignore alignment locks, we can set the new field (`m_Vector`) directly, though it might cause weird behavior. 🤷

Reply:
It seems like it should not be hard to get this right.

convert the polar coordinates to cartesian 
x = r*cos(dir)
y = r*sin(dir)
apply the shift to x or y based on the direction, and cap it as necessary 
then convert back 
r = sqrt(x^2 + y^2)
dir = atan(y/x) or use atan2